### PR TITLE
added url on/off and css styles

### PR DESCRIPTION
**Pull request by DTabidze**

### CODE REVIEW

File: apps/url-shortener/index.html [hunk 1]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
         <tr>
             <th>Name</th>
             <th>Address</th>
-            <th>Owner</th>
+            <!-- <th>Owner</th> -->
```
Comment (giolekva) [inline apps/url-shortener/index.html:29]: remove line

File: apps/url-shortener/index.html [hunk 2]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
             <th>Active</th>
         </tr>
         {{ range .NamedAddresses }}
         <tr>
             <td><a href="{{ .Name }}" target="_blank">{{ .Name }}</a></td>
             <td>{{ .Address }}</td>
-            <td>{{ .OwnerId }}</td>
-            <td>{{ .Active }}</td>
+            <!-- <td>{{ .OwnerId }}</td> -->
+            <!-- <td>{{ .Active }}</td> -->
```
Comment (giolekva) [inline apps/url-shortener/index.html:37]: remove

File: apps/url-shortener/index.html [hunk 3]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+            <td>
+                <label for="switch">
```
Comment (giolekva) [inline apps/url-shortener/index.html:39]: You don't need label tag here

File: apps/url-shortener/index.html [hunk 4]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+                    <input type="checkbox" id="switch-{{ .Name }}" name="switch" role="switch" {{ if .Active }}checked{{ end }}
```
Comment (giolekva) [inline apps/url-shortener/index.html:40]: You can remove id and name properties, they are not used anywhere

Comment (giolekva) [inline apps/url-shortener/index.html:40]: Add space after comma and spaces after/before curly braces: '{{ .Name }}', {{ not .Active }}

File: apps/url-shortener/index.html [hunk 5]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
     </table>
 </body>
-
+<script>
+    function toggle(name,status) {
+        const data = {
+            "name": name,
+            "active": status
+        }
+        console.log("DATA: ", data)
```
Comment (giolekva) [inline apps/url-shortener/index.html:55]: remove

File: apps/url-shortener/index.html [hunk 6]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data)
+        })
+        .then(response => {
+            if (response.ok) {
+                window.location.reload();
+            }
+        })
+        // .then (data => console.log("success: ", data))
```
Comment (giolekva) [inline apps/url-shortener/index.html:68]: remove

File: apps/url-shortener/index.html [hunk 7]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+        .catch((error) => {
+            console.error('Error:', error);
+        });
+        // console.log(name,status," CHANGED!")
```
Comment (giolekva) [inline apps/url-shortener/index.html:72]: remove

File: apps/url-shortener/index.html [hunk 8]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
         {{ end }}
     </table>
 </body>
-
+<script>
+    function toggle(name,status) {
+        const data = {
+            "name": name,
+            "active": status
+        }
```
Comment (giolekva) [inline apps/url-shortener/index.html:54]: add ; at the end of expression

File: apps/url-shortener/index.html [hunk 9]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+                    >
+                </label>
+            </td>
         </tr>
         {{ end }}
     </table>
 </body>
-
+<script>
+    function toggle(name,status) {
```
Comment (giolekva) [inline apps/url-shortener/index.html:50]: add space after comma

File: apps/url-shortener/index.html [hunk 10]
```diff
@@ -25,18 +26,50 @@ <h2>Named Addresses:</h2>
...
+                    onclick="toggle('{{ .Name }}',{{not .Active}})"
+                    >
+                </label>
+            </td>
         </tr>
         {{ end }}
     </table>
 </body>
-
+<script>
```
Comment (giolekva) [inline apps/url-shortener/index.html:49]: Add `type="application/javascript"` to script tag

File: apps/url-shortener/main.go [hunk 11]
```diff
@@ -151,12 +150,18 @@ func renderHTML(w http.ResponseWriter, r *http.Request, tpl *template.Template,
...
+	return "tabo", nil
+}
+
 type Server struct {
 	store Store
 }
 
 func (s *Server) Start() {
 	http.HandleFunc("/", s.handler)
+	http.HandleFunc("/api/", s.togglehandler)
```
Comment (giolekva) [inline apps/url-shortener/main.go:164]: rename this to `/api/update`

File: apps/url-shortener/main.go [hunk 12]
```diff
@@ -174,10 +179,15 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 				cn = generateRandomURL()
 			}
 			// check if custom exists
+			loggedInUser, err := getLoggedInUser(r)
+			if err != nil {
+				http.Error(w, "User Not Logged In", http.StatusUnauthorized)
+				return
+			}
```
Comment (giolekva) [inline apps/url-shortener/main.go:186]: You are fetching logged in user twice in this function, you can deduplicate them by moving this block at the beginning of the function

File: apps/url-shortener/main.go [hunk 13]
```diff
@@ -228,6 +248,39 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 	renderHTML(w, r, tmpl, pageVariables)
 }
 
+type UpdateRequest struct {
+	Name   string `json:"name"`
+	Active bool   `json:"active"`
+}
+
+func (s *Server) togglehandler(w http.ResponseWriter, r *http.Request) {
```
Comment (giolekva) [inline apps/url-shortener/main.go:256]: Rename to `toggleHandler`

File: apps/url-shortener/main.go [hunk 14]
```diff
@@ -228,6 +248,39 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
...
+	var data UpdateRequest
+	if r.Method == http.MethodPost {
```
Comment (giolekva) [inline apps/url-shortener/main.go:258]: This function should return an error if non `POST` method is invoked on `/api` endpoint

File: apps/url-shortener/main.go [hunk 15]
```diff
@@ -228,6 +248,39 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
...
+		namedAddress, err := s.store.Get(data.Name)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to get named_address for name %s", data.Name), http.StatusInternalServerError)
+			return
+		}
+		loggedInUser, err := getLoggedInUser(r)
+		if err != nil {
+			http.Error(w, "User Not Logged In", http.StatusUnauthorized)
+			return
+		}
```
Comment (giolekva) [inline apps/url-shortener/main.go:272]: I'd check logged in user before decoding request

**Review by giolekva — COMMENTED** [review]
Overall looks good, I left many comments but all of them are minor and stylistic

File: apps/url-shortener/index.html (hunk 1 above)
Comment (DTabidze) [reply apps/url-shortener/index.html:29]: Done

File: apps/url-shortener/index.html (hunk 2 above)
Comment (DTabidze) [reply apps/url-shortener/index.html:37]: Done

File: apps/url-shortener/index.html (hunk 3 above)
Comment (DTabidze) [reply apps/url-shortener/index.html:39]: Done

File: apps/url-shortener/index.html (hunk 4 above)
Comment (DTabidze) [reply apps/url-shortener/index.html:40]: Done

File: apps/url-shortener/main.go (hunk 11 above)
Comment (DTabidze) [reply apps/url-shortener/main.go:164]: Done

File: apps/url-shortener/main.go (hunk 12 above)
Comment (DTabidze) [reply apps/url-shortener/main.go:186]: Done

File: apps/url-shortener/main.go (hunk 13 above)
Comment (DTabidze) [reply apps/url-shortener/main.go:256]: Done

File: apps/url-shortener/main.go (hunk 14 above)
Comment (DTabidze) [reply apps/url-shortener/main.go:258]: Done

File: apps/url-shortener/main.go (hunk 15 above)
Comment (DTabidze) [reply apps/url-shortener/main.go:272]: Done

File: apps/url-shortener/index.html (hunk 10 above)
Comment (DTabidze) [reply apps/url-shortener/index.html]: Done

File: apps/url-shortener/index.html (hunk 9 above)
Comment (DTabidze) [reply apps/url-shortener/index.html]: Done

File: apps/url-shortener/index.html (hunk 8 above)
Comment (DTabidze) [reply apps/url-shortener/index.html]: Done

File: apps/url-shortener/index.html (hunk 5 above)
Comment (DTabidze) [reply apps/url-shortener/index.html]: Done

**Pull request merged**